usr/lib/common/pkcs_utils.c

```diff
@@ -0,0 +1,442 @@
+/*
+ * COPYRIGHT (c) International Business Machines Corp. 2020
+ *
+ * This program is provided under the terms of the Common Public License,
+ * version 1.0 (CPL-1.0). Any use, reproduction or distribution for this
+ * software constitutes recipient's acceptance of CPL-1.0 terms which can be
+ * found in the file LICENSE file or at
+ * https://opensource.org/licenses/cpl1.0.php
+ */
+
+/*
+ * Some routines that are shared between the pkcs utilities in usr/sbin.
+ */
+
+#define _GNU_SOURCE
+#include <dlfcn.h>
+#include <errno.h>
+#include <openssl/evp.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <termios.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <dirent.h>
+#include <openssl/evp.h>
+#include <pkcs11types.h>
+
+#include "defs.h"
+#include "host_defs.h"
+#include "pkcs_utils.h"
+
+pkcs_trace_level_t trace_level = TRACE_LEVEL_NONE;
```

Comment:
So if this source file is to be included in the tools as well as in OCK base code (common code or tokens), then this global variable should be extern here, and should be defined in the tool code, and somewhere in OCK common code.

Reply:
Right, done.